android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java

```diff
@@ -358,16 +358,20 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 return true;
 
             case R.id.action_share:
+                StringBuilder speakers  = new StringBuilder("");
+                for(int i = 0; i < session.getSpeakers().size(); i++){
+                    speakers.append(((i == 0)?(""):(", ")) + session.getSpeakers().get(i).getName());
+                }
```

Comment:
Where are you converting it back to the String ? After the for loop , convert it to a String using ```toString()```.

Reply:
No need for `new StringBuilder("")`, `new StringBuilder()` will do

Reply:
Rename `speakers` to `speakersNameBuilder`

Reply:
Use for each loop instead

```
for (Speaker speaker: speakers) {
    // append
}

// remove the last character if builder is not empty
```